### PR TITLE
Device reduction isa

### PR DESCRIPTION
**Pull request by liujing2**
This passed CI. Please help review, thanks!
@sameo

### CODE REVIEW

**Comment by liujing2** [comment]
RTC need some big refactor, because some initialization codes couldn’t justify the platform you choose to use, for example when we both compile pc and q35 into the nemu binary but launch a q35 guest. 
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
void pc_basic_device_init(ISABus *isa_bus, qemu_irq *gsi, ISADevice **rtc_state, bool create_fdctrl, bool has_pit)
{
….
    if (piix_enabled()) {			//return true when using q35 but --enable-piix
        *rtc_state = mc146818_rtc_init(isa_bus, 2000, rtc_irq);
        qemu_register_boot_set(pc_boot_set, *rtc_state);
    }
    ….
}
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

**Comment by liujing2** [comment]
One of the patches changes the CI tests. Will this automatic CI test in backend do the same change?

**Comment by sameo** [comment]
> Will this automatic CI test in backend do the same change?

Yes, it will.

**Comment by liujing2** [comment]
@sameo  I locally passed the CI for both q35/pc unsafe/safe images on x86. Don't know why it reported failed here?

liujing@he:~/nemu$ sudo -E ~/nemu/tools/CI/minimal_ci.sh -hypervisor ~/build-x86_64/x86_64-softmmu/qemu-system-x86_64 -piix
[sudo] password for liujing:
Testing safe images
Testing clear-22180-cloud.img qcow2 uefi x86_64_pc https://download.clearlinux.org/releases/22180/clear/clear-22180-cloud.img.xz :
SUCCESS: SSH connectivity verified
Testing CentOS-6-x86_64-GenericCloud-1802.qcow2 qcow2 seabios x86_64_pc https://cloud.centos.org/centos/6/images/CentOS-6-x86_64-GenericCloud-1802.qcow2 :
SUCCESS: SSH connectivity verified
Testing coreos_production_openstack_image.img qcow2 seabios x86_64_pc https://stable.release.core-os.net/amd64-usr/current/coreos_production_openstack_image.img.bz2 :
SUCCESS: SSH connectivity verified
Testing bionic-server-cloudimg-amd64.img qcow2 seabios x86_64_pc http://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img :
SUCCESS: SSH connectivity verified
Testing CentOS-7-x86_64-GenericCloud-1802.qcow2 qcow2 seabios x86_64_pc https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1802.qcow2 :
SUCCESS: SSH connectivity verified
Testing CentOS-Atomic-Host-7-GenericCloud.qcow2 qcow2 seabios x86_64_pc http://cloud.centos.org/centos/7/atomic/images/CentOS-Atomic-Host-7-GenericCloud.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-AtomicHost-28-1.1.x86_64.qcow2 qcow2 seabios x86_64_pc https://archives.fedoraproject.org/pub/fedora/linux/releases/28/AtomicHost/x86_64/images/Fedora-AtomicHost-28-1.1.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-Cloud-Base-27-1.6.x86_64.qcow2 qcow2 seabios x86_64_pc https://archives.fedoraproject.org/pub/fedora/linux/releases/27/CloudImages/x86_64/images/Fedora-Cloud-Base-27-1.6.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-Cloud-Base-28-1.1.x86_64.qcow2 qcow2 seabios x86_64_pc https://archives.fedoraproject.org/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing coreos_production_openstack_image.img qcow2 seabios x86_64_q35 https://stable.release.core-os.net/amd64-usr/current/coreos_production_openstack_image.img.bz2 :
SUCCESS: SSH connectivity verified
Testing CentOS-7-x86_64-GenericCloud-1802.qcow2 qcow2 seabios x86_64_q35 https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1802.qcow2 :
SUCCESS: SSH connectivity verified
Testing CentOS-Atomic-Host-7-GenericCloud.qcow2 qcow2 seabios x86_64_q35 http://cloud.centos.org/centos/7/atomic/images/CentOS-Atomic-Host-7-GenericCloud.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-AtomicHost-28-1.1.x86_64.qcow2 qcow2 seabios x86_64_q35 https://archives.fedoraproject.org/pub/fedora/linux/releases/28/AtomicHost/x86_64/images/Fedora-AtomicHost-28-1.1.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-Cloud-Base-27-1.6.x86_64.qcow2 qcow2 seabios x86_64_q35 https://archives.fedoraproject.org/pub/fedora/linux/releases/27/CloudImages/x86_64/images/Fedora-Cloud-Base-27-1.6.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing Fedora-Cloud-Base-28-1.1.x86_64.qcow2 qcow2 seabios x86_64_q35 https://archives.fedoraproject.org/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.qcow2 :
SUCCESS: SSH connectivity verified

liujing@he:~/nemu$ sudo -E ~/nemu/tools/CI/minimal_ci.sh -hypervisor ~/build-x86_64/x86_64-softmmu/qemu-system-x86_64 -piix -unsafe
Testing unsafe images
Testing ubuntu-12.04-server-cloudimg-amd64-disk1.img qcow2 seabios x86_64_pc http://cloud-images-archive.ubuntu.com/releases/precise/release-20170502/ubuntu-12.04-server-cloudimg-amd64-disk1.img :
SUCCESS: SSH connectivity verified
Testing ubuntu-11.10-server-cloudimg-amd64-disk1.img qcow2 seabios x86_64_pc http://cloud-images-archive.ubuntu.com/releases/oneiric/release-20130509/ubuntu-11.10-server-cloudimg-amd64-disk1.img :
SUCCESS: SSH connectivity verified
Testing core-stable-amd64-disk1.img qcow2 seabios x86_64_pc https://cloud-images.ubuntu.com/snappy/15.04/core/stable/current/core-stable-amd64-disk1.img :
SUCCESS: SSH connectivity verified
Testing Fedora-Cloud-Base-20141203-21.x86_64.qcow2 qcow2 seabios x86_64_pc https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/21/Cloud/Images/x86_64/Fedora-Cloud-Base-20141203-21.x86_64.qcow2 :
SUCCESS: SSH connectivity verified
Testing core-stable-amd64-disk1.img qcow2 seabios x86_64_q35 https://cloud-images.ubuntu.com/snappy/15.04/core/stable/current/core-stable-amd64-disk1.img :
SUCCESS: SSH connectivity verified

File: accel/kvm/kvm-all.c [hunk 1]
```diff
@@ -1216,10 +1216,6 @@ int kvm_irqchip_update_msi_route(KVMState *s, int virq, MSIMessage msg,
         return 0;
     }
 
-    if (!kvm_irqchip_in_kernel()) {
```
Comment (sameo) [inline accel/kvm/kvm-all.c:1219]: It seems we're assuming that we have an in kernel interrupt controller implementation all over that patch. This seems contradictory with the previous patch where we're going with split irq all the way.
Could you please elaborate a little further?

**Comment by sameo** [comment]
@liujing2 Could you please explain why we need split irq in order to start reducing the number of ISA devices?

**Comment by liujing2** [comment]
@sameo Split irqchip mode is not intended for reducing the number of ISA. It is just a request from Manohar that we need split mode now. So I revert the kernel_irqchip=on mode patches and changes it to kernel_irqchip=split.

File: accel/kvm/kvm-all.c (hunk 1 above)
Comment (liujing2) [reply accel/kvm/kvm-all.c:1219]: Assuming that we use kernel_irqchip=split, the codes around this function can be removed.

**Pull request closed without merging**